torch/_inductor/codegen/triton.py

```diff
@@ -822,13 +822,15 @@ def load(self, name: str, index: sympy.Expr):
         indirect_indexing = self.is_indirect_indexing(index)
         index, mask_vars, mask = self.indexing(index)
 
-        if "rmask" in mask:
-            # This eviction policy heuristic is untested.
-            # ptillet suggested we should try only doing this for
-            # the first N-1 loops and not for the final loop.
-            ep = ", eviction_policy='evict_last'"
-        else:
-            ep = ""
+        # Keep the variable in cache if we are going to reuse it
+        # TODO(lezcano) We could potentially do better
+        # https://github.com/pytorch/pytorch/pull/91316#issuecomment-1364680622
+        ep = (
+            ", eviction_policy='evict_last'"
+            if name not in self.current_node.last_usage
```

Comment:
IIUC this tests if there will be future kernels that re-use the buffer being loaded. If so, this seems insufficient to me because a reduction kernel such as softmax may load the same memory location multiple times, but if fused into a single kernel this test doesn't keep it in cache.

Reply:
Not quite, see how this PR modifies for how that `last_usage` node is set for nodes within a `FusedSchedulerNode`. But I think the issue similar to that one. I think this PR is not setting this correctly for nodes that are not fused.